_posts/2016-04-26-distributed-game-of-life-in-elixir.md

```diff
@@ -0,0 +1,1065 @@
+  end
+
+  def start_game(speed \\ @game_speed) do
+    GenServer.call(@name, {:start_game, speed})
+  end
+
+  def stop_game do
+    GenServer.call(@name, :stop_game)
+  end
+
+  def change_speed(speed) do
+    stop_game
+    start_game(speed)
+  end
+end
+{% endhighlight %}
+
+As you can see we use `GenServer` behaviour in our module. I require also Logger as we would like to print some info to the STDOUT.
+
+In `start_link/1` function we start new `GenServer`. When our generic server was started as a first process in the cluster then it becomes master process. In the case when there is already running process with globally registered name `{:global, __MODULE__}` we log info that our process will be a slave process with a reference to existing PID on another node in the cluster.
+
+As you see we store global name for our server in attribute `@name`. We use another attribute `@game_speed` for default game speed which is 1000 miliseconds.
+
+In our public interface, we have `alive_cells/1` function which returns the list of alive cells. Basically, it's the current state of the game (alive cells on the board). This function calls `GenServer` with registered `@name` and request `:alive_cells`. We need to implement `handle_call/3` function for this type of request (`:alive_cells`).
+
+There is another public function `generation_counter/1` which returns how many generations was already processed by board server.
+
+The `state/1` function returns state that is held by our generic server. The state is represented as the tuple with 3 values like alive cells, TRef (time reference - we want to regenerate board every second) and generation counter. TRef is very internal thing for board server so we won't return this to outside world. That's why we will return just alive cells and generation counter. You will see it later in implementation for `handle_call(:state, _from, state)`.
+
+You can use `set_alive_cells/1` function in the case when you want to override current list of alive cells with a new list.
+
+The `add_cells/1` function will be very usefull as we want to be able to add new cells or figures to the board. For instance we may want to add a blinker pattern to existing game. You will learn more about patterns later.
+
+<a href="https://en.wikipedia.org/wiki/File:Game_of_life_blinker.gif"><img src="/images/blog/posts/distributed-game-of-life-in-elixir/blinker.gif" /></a>
+
+We can manually force game to calculate next generation of cells with `tick/1` function.
+
+The `start_game/1` function is responsible for starting a new timer which calls every second a `tick/1` function. Thanks to that our game will update list of alive cells with specified interval which is `@game_speed`.
+
+The last 2 functions are `stop_game/1` and `change_speed/1` which just restart the game and starts a new one with provided speed.
+
+Now you can take a look how above functions are working exactly because they are calling server callbacks.
```

Comment:
functions are working. (I think you can remove the rest of this sentence, or divide it to two separete sentences)